src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php

```diff
@@ -272,6 +272,7 @@ public function applyDefaults(ContainerBuilder $container, $serviceId, array $at
 
         $defaultAddServices = [
             'model_manager' => sprintf('sonata.admin.manager.%s', $managerType),
+            'data_source' => sprintf('sonata.admin.data_source.%s', $managerType),
```

Comment:
```suggestion
            'data_source' => sprintf('?sonata.admin.data_source.%s', $managerType),
```

Reply:
This doesn't work. 

The failure is: The service "Sonata\AdminBundle\Tests\App\Admin\FooAdmin" has a dependency on a non-existent service "?sonata.admin.data_source.test"

Reply:
My bad, I miss it with ` public static function getSubscribedServices()`